python/cm/models.py

```diff
@@ -211,32 +221,65 @@ class ADCMEntity(ADCMModel):
     config = models.OneToOneField(ObjectConfig, on_delete=models.CASCADE, null=True)
     state = models.CharField(max_length=64, default='created')
     _multi_state = models.JSONField(default=dict, db_column='multi_state')
-    issue = models.JSONField(default=dict)
-    concern = models.ManyToManyField('ConcernItem', blank=True, related_name='%(class)s_entities')
+    _issues = models.ManyToManyField(
+        'ConcernItem', blank=True, db_column='issues', related_name='%(class)s_sources'
+    )
+    concerns = models.ManyToManyField('ConcernItem', blank=True, related_name='%(class)s_entities')
 
     class Meta:
         abstract = True
 
+    class IssueType(Enum):
+        Config = 'config'
+        RequiredService = 'required_service'
+        RequiredImport = 'required_import'
+        HostComponent = 'host_component'
+
     @property
     def is_locked(self) -> bool:
-        return self.concern.exists()
+        """Check if actions could be run over entity"""
+        return self.concerns.exists()
 
-    def add_to_concern(self, item: 'ConcernItem') -> None:
+    def add_to_concerns(self, item: 'ConcernItem') -> None:
+        """Attach entity to ConcernItem to keep up with it"""
         if not item or getattr(item, 'id', None) is None:
             return
 
-        if item not in self.concern.all():
-            self.concern.add(item)
+        if item in self.concerns.all():
+            return
+
+        self.concerns.add(item)
 
-    def remove_from_concern(self, item: 'ConcernItem') -> None:
+    def remove_from_concerns(self, item: 'ConcernItem') -> None:
+        """Detach entity from ConcernItem when it outdated"""
         if not item or not hasattr(item, 'id'):
             return
 
-        if item in self.concern.all():
-            self.concern.remove(item)
+        if item not in self.concerns.all():
+            return
+
+        self.concerns.remove(item)
+
+    def gen_issue_name(self, issue_name: IssueType) -> str:
+        """Create unique issue name"""
+        return f'{issue_name.value} issue on {self.prototype.type} {self.pk}'
+
+    def add_issue(self, issue: 'ConcernItem') -> None:
+        """Add issue to its source"""
+        for _ in issue.source_objects:
+            raise AdcmEx('ISSUE_INTEGRITY_ERROR')
+
+        self._issues.add(issue)
+
+    def get_issue(self, issue_type: IssueType) -> Optional['ConcernItem']:
+        """Get object's issue of specified type or None"""
+        issue_name = self.gen_issue_name(issue_type)
+        for issue in self._issues.filter(name=issue_name):
```

Comment:
Are we expected exactly one issue here? Or return only the first one?

Reply:
one or zero, more than one of same type (and same name) is imposible because of constraint